cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -835,6 +835,48 @@ const char* replace_utf8_utf8_utf8(gdv_int64 context, const char* text,
                                              out_len);
 }
 
+FORCE_INLINE
+const char* binary_string(gdv_int64 context, const char* text, gdv_int32 text_len,
+                          gdv_int32* out_len) {
+  gdv_binary ret =
+      reinterpret_cast<gdv_binary>(gdv_fn_context_arena_malloc(context, text_len));
+
+  if (ret == nullptr) {
+    gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
```

Comment:
function that can return errors needs to set that in the function definition : NativeFunction::kNeedsContext | NativeFunction::kCanReturnErrors

Reply:
added

Reply:
i still see the same above?

Reply:
Resolving based on offline discussion.